### PR TITLE
fix: bugs surrounding edit-classroom modal

### PR DESCRIPTION
**Pull request by jfdoming**
## Notion ticket link
No ticket - bugs I noticed in the course of using the platform

## Implementation description
- Fetch missing `startDate` field
- Fix classroom page constantly rerendering
- Reset form data to default values rather than empty state
- Hydrate default values on classrooms page
- Refetch correct queries upon update

## Steps to test
1. Open the modals on the classrooms page and classroom drilldown page several times.
2. Verify that the values are reset to default (current) values each time.

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by carissa-tang — COMMENTED** [review]
could you look into this error i get when adding a new classroom?
```
"Validation failed: grades.0: `Grade 3` is not a valid enum value for path `grades.$`."
```
image

The class ends up being created, but there's an error message

**Comment by jfdoming** [comment]
> could you look into this error i get when adding a new classroom?
> 
> ```
> "Validation failed: grades.0: `Grade 3` is not a valid enum value for path `grades.$`."
> ```
> 
> image
> The class ends up being created, but there's an error message

Hmm, not able to repro this... Maybe let's chat about it in the sync today

File: frontend/src/utils/GeneralUtils.ts [hunk 1]
```diff
@@ -94,3 +96,11 @@ export const getLetterFromNumber = (number: number): string => {
 
   return firstLetter + secondLetter;
 };
+
+export const getQueryName = (query: DocumentNode): string => {
+  const name = getOperationAST(query)?.name?.value;
+  if (!name) {
+    throw new Error("Query name not found. This is probably a bug.");
```
Comment (carissa-tang) [inline frontend/src/utils/GeneralUtils.ts:103]: is there some info about the query we can add into the error message? or the query itself lol

**Review by carissa-tang — APPROVED** [review]
nice!

File: frontend/src/utils/GeneralUtils.ts (hunk 1 above)
Comment (jfdoming) [reply frontend/src/utils/GeneralUtils.ts:103]: good point lol

edit: we decided not to bc there's nothing useful to print

**Pull request merged**